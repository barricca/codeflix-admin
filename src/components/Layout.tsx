import { Box, Container } from "@mui/material";

export function Layout({ children }: { readonly children: React.ReactNode }) {
	return (
		<Box>
			<Container maxWidth="lg" sx={{ color: "white", mt: 4, mb: 4 }}>
				{children}
			</Container>
		</Box>
	);
}
