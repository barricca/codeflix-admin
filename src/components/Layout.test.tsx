import { Typography } from "@mui/material";
import { render } from "@testing-library/react";
import { Layout } from "./Layout";

describe("Layout", () => {
	it("should render Layout with children correctly", () => {
		const { asFragment } = render(
			<Layout>
				<Typography variant="h1">Test Title</Typography>
			</Layout>,
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
