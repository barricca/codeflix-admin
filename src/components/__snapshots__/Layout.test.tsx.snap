// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render Layout with children correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1km29jt-MuiContainer-root"
    >
      <h1
        class="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
      >
        Test Title
      </h1>
    </div>
  </div>
</DocumentFragment>
`;
